v2.2/query-behavior-troubleshooting.md

```diff
@@ -1,31 +1,94 @@
-- Your application has a bug. It's always worthwhile to check and double-check your application’s logic before filing an issue. That said, you can always [reach out for support](support-resources.html).
-- CockroachDB has a bug. Please [file an issue](file-an-issue.html).
+-  [Indexes best practices](https://www.cockroachlabs.com/docs/stable/performance-best-practices-overview.html#indexes-best-practices)
 
-## Performance issues
+## Query is always slow
 
-If queries are taking longer than expected to process, there are a few things you can check into:
+To detect whether your cluster has slow queries, check the [service latency graph](https://www.cockroachlabs.com/docs/stable/admin-ui-sql-dashboard.html#service-latency-sql-99th-percentile) and the [CPU graph](https://www.cockroachlabs.com/docs/v2.1/admin-ui-hardware-dashboard.html#cpu-percent). If the graphs show latency spikes or CPU usage spikes, it might be an indication of slow queries in your cluster.
 
-- Review your deployment's monitoring. General network latency or partitioning events can affect query response times.
+Once you determine that you do have slow queries in your cluster, use the [Statements page](https://www.cockroachlabs.com/docs/stable/admin-ui-statements-page.html) to identify the high latency [SQL statements](https://www.cockroachlabs.com/docs/stable/sql-statements.html). To view the Statements page, [access the Admin UI](https://www.cockroachlabs.com/docs/stable/admin-ui-access-and-navigate.html#access-the-admin-ui) and then click Statements on the left.
 
-- [Identify and cancel long-running queries](manage-long-running-queries.html).
+You can then use the [Statements Details](https://www.cockroachlabs.com/docs/stable/admin-ui-statements-page.html#statement-details-page) page to drill down to individual statements. You can also use [`EXPLAIN ANALYZE`](https://www.cockroachlabs.com/docs/stable/explain-analyze.html) statement, which executes a SQL query and returns a physical query plan with execution statistics. Query plans provide information around SQL execution, which can be used to troubleshoot slow queries by figuring out where time is being spent, how long a processor (i.e., a component that takes streams of input rows and processes them according to a specification) is not doing work, etc.
 
-- [Turn on SQL logging](#sql-logging).
+If you need help interpreting the output of the `EXPLAIN ANALYZE` statement, contact us.
 
-If you're still unable to determine why the query executes slowly, please [file an issue](file-an-issue.html).
+## Query is sometimes slow
+
+If the query performance is irregular:
+
+1.  Run [`SHOW TRACE`](https://www.cockroachlabs.com/docs/stable/show-trace.html) for the query twice: once when the query is performing as expected and once when the query is slow.
+
+2.  Contact us to analyze the outputs of the `SHOW TRACE` command.
+
+## Cancelling running queries
```

Comment:
This could use some context, like when/why you might do this vs. the other troubleshooting steps.